File: algorithms/twoSum.js
function twoSum(target, array) {
	let i = 0;
	let j = array.length - 1;
	array.sort((a, b) => a - b);
	while (i <= j) {
		let sum = array[i] + array[j];
		if (sum > target) {
			j -= 1;
		} else if (sum < target) {
			i += 1;
		} else {
			return [array[i], array[j]];
		}
	}
	return 'No pairs';
}

module.exports = twoSum;

File: algorithms/Python/twosum.py
def two_sum(arr, target):
    hash = {}
    for index, num in enumerate(arr):
        diff = target - num
        print("mydiff", diff)
        if diff in hash:
            print(hash[diff], index)
        hash[num] = index

    print(hash)


two_sum([2, 7, 11, 15], 9)

# [2,7,11,15]
# 9
